docs/tutorial.md

```diff
@@ -0,0 +1,182 @@
+- Generate the manifests using `config/default`
+- Comment out `ClusterRole` and `ClusterRoleBinding`. 
+- Create a `ClusterRole` template in a `ScopeTemplate` and associate it with a `ScopeInstance`
+- Run the operator, the `memcached-operator` will start complaining about `RBAC`.
+- Run the `oria-operator`, it will create `ClusterRole` and `ClusterRoleBinding`. This will make sure that `memcached-operator` will not complain about `RBAC`. 
+
+### Install and run oria-operator
+
+Clone the [github repo](https://github.com/operator-framework/oria-operator.git) and  run the below commands on the root of the directory.
+
+```
+$ make install
+
+$ make run
+```
+
+### Create a new project
+
+Use the [Operator SDK](https://sdk.operatorframework.io/docs/installation/) CLI to create a new memcached-operator from the [tutorial](https://sdk.operatorframework.io/docs/building-operators/golang/tutorial/).
+
+
+```sh
+mkdir -p $HOME/projects/memcached-operator
+cd $HOME/projects/memcached-operator
+# we'll use a domain of example.com
+# so all API groups will be <group>.example.com
+operator-sdk init --domain example.com --repo github.com/example/memcached-operator
+```
+
+### Create a new API and Controller 
+
+```sh
+$ operator-sdk create api --group cache --version v1alpha1 --kind Memcached --resource --controller
+Writing scaffold for you to edit...
+api/v1alpha1/memcached_types.go
+controllers/memcached_controller.go
+...
+```
+
+Define the API for the Memcached Custom Resource(CR) by modifying the Go type definitions at `api/v1alpha1/memcached_types.go` to have the following spec and status:
+
+```
+// MemcachedSpec defines the desired state of Memcached
+type MemcachedSpec struct {
+	//+kubebuilder:validation:Minimum=0
+	// Size is the size of the memcached deployment
+	Size int32 `json:"size"`
+}
+
+// MemcachedStatus defines the observed state of Memcached
+type MemcachedStatus struct {
+	// Nodes are the names of the memcached pods
+	Nodes []string `json:"nodes"`
+}
+```
+
+### Now create docker image and push to the dockerhub
+
+Your Makefile composes image tags either from values written at project initialization or from the CLI. In particular, IMAGE_TAG_BASE lets you define a common image registry, namespace, and partial name for all your image tags. Update this to another registry and/or namespace if the current value is incorrect. Afterwards you can update the IMG variable definition like so:
+
+```
+-IMG ?= controller:latest
++IMG ?= $(IMAGE_TAG_BASE):$(VERSION)
+```
+
+Once done, you do not have to set IMG or any other image variable in the CLI. The following command will build and push an operator image tagged as example.com/memcached-operator:v0.0.1 to Docker Hub:
+
+```
+make docker-build docker-push
+```
+
+### Generating CRD manifests 
+
+```sh
+$ mkdir manifest
+$ kustomize build config/default > manifest/manifest.yaml
```

Comment:
1. I would split this into 2 code blocks or combine the commands with `&&`
2. Where do I get `kustomize` from? (`make kustomize` might be a nice step to include here because it will download it to `bin/kustomize`)

Reply:
yeah, I will add as a prerequisite at the top.